### PR TITLE
Adding Gaussian Beam summation

### PR DESCRIPTION
**Pull request by DavidLanders95**

### CODE REVIEW

**Comment by uellue** [comment]
It looks like Microscope Calibration works fine after rebasing #4 to this and adapting to `run_iter()` changes. I can't really say much about the rest! From my perspective one can merge this.

**Comment by uellue** [comment]
Also turned out that jax is [aggressively casting down to float32](https://docs.jax.dev/en/latest/notebooks/Common_Gotchas_in_JAX.html#double-64bit-precision). I've sprinkled `import jax; jax.config.update("jax_enable_x64", True)` all over the code and start requesting float64 explicitly where necessary. That seems to be an important part of resolving this!

PS: Success! With float64 and manually specified inverse we now pass `atol` and `rtol` of `1e-12` for the linearized solution!

```python
    x, residuals, rank, s = np.linalg.lstsq(input_samples, output_samples)

    assert len(residuals) == rank
    # Confirm that the solution is exact, in particular that
    # the model is linear
    assert np.allclose(residuals, 0., rtol=1e-12, atol=1e-12)
    assert rank == 5

    for i in range(len(input_samples)):
        assert np.allclose(
            output_samples[i],
            input_samples[i] @ x,
            rtol=1e-12,
            atol=1e-12
        )
    return x
```

**Comment by DavidLanders95** [comment]
> Also turned out that jax is [aggressively casting down to float32](https://docs.jax.dev/en/latest/notebooks/Common_Gotchas_in_JAX.html#double-64bit-precision). I've sprinkled `import jax; jax.config.update("jax_enable_x64", True)` all over the code and start requesting float64 explicitly where necessary. That seems to be an important part of resolving this!
> 
> PS: Success! With float64 and manually specified inverse we now pass `atol` and `rtol` of `1e-12` for the linearized solution!
> 
> ```python
>     x, residuals, rank, s = np.linalg.lstsq(input_samples, output_samples)
> 
>     assert len(residuals) == rank
>     # Confirm that the solution is exact, in particular that
>     # the model is linear
>     assert np.allclose(residuals, 0., rtol=1e-12, atol=1e-12)
>     assert rank == 5
> 
>     for i in range(len(input_samples)):
>         assert np.allclose(
>             output_samples[i],
>             input_samples[i] @ x,
>             rtol=1e-12,
>             atol=1e-12
>         )
>     return x
> ```

For me I by default think it's necessary to turn on this feature, since the scales, and the disparity in scales we work at demand it. 
We could just enable it by default package wide?

I have been just enabling float64 at the top of any notebook. Then I don't need to explicitly make anything float64, and it will automatically do so (at least I haven't found a case where it used float32 when I didn't want it to).

**Comment by DavidLanders95** [comment]
> It looks like Microscope Calibration works fine after rebasing #4 to this and adapting to `run_iter()` changes. I can't really say much about the rest! From my perspective one can merge this.

Ok, after adding more tests and lots of examples, I will merge now.

**Comment by uellue** [comment]
> For me I by default think it's necessary to turn on this feature, since the scales, and the disparity in scales we work at demand it.
> We could just enable it by default package wide?

Yes! Microscope Calibration already enables by default, validates the precision and accuracy of linearizations, and will raise an error if it detects discrepancies that exceed tolerances achievable with float64 .
